### PR TITLE
Postcopy

### PR DESCRIPTION
**Pull request by dagrh**
Now the avocado changes have gone in, it's time for the tp changes.
With this we get a simple 'migrate.default.tcp.with_post_copy' that normally does a handful of postcopy requests.

### CODE REVIEW

File: qemu/tests/migration.py [hunk 1]
```diff
@@ -206,17 +209,29 @@ def check_dma():
...
+                    vm.migrate(mig_timeout, mig_protocol, mig_cancel_delay,
+                               offline, check,
+                               migration_exec_cmd_src=mig_exec_cmd_src,
+                               migration_exec_cmd_dst=mig_exec_cmd_dst,
+                               migrate_capabilities=capabilities,
+                               mig_inner_funcs=inner_funcs,
+                               env=env)
+                except qemu_monitor.MonitorNotSupportedMigCapError as e:
+                    logging.info("Unable to access capability: %s" % e)
+                    raise exceptions.TestSkipError(e)
```
Comment (balamuruhans) [inline qemu/tests/migration.py]: we have stopped using `exceptions` instead `test.cancel()` would help us here.

File: qemu/tests/migration.py [hunk 2]
```diff
@@ -206,17 +209,29 @@ def check_dma():
...
+                try:
+                    vm.migrate(mig_timeout, mig_protocol, mig_cancel_delay,
+                               offline, check,
+                               migration_exec_cmd_src=mig_exec_cmd_src,
+                               migration_exec_cmd_dst=mig_exec_cmd_dst,
+                               migrate_capabilities=capabilities,
+                               mig_inner_funcs=inner_funcs,
+                               env=env)
+                except qemu_monitor.MonitorNotSupportedMigCapError as e:
+                    logging.info("Unable to access capability: %s" % e)
```
Comment (balamuruhans) [inline qemu/tests/migration.py]:
```python
logging.info("Unable to access capability: %s", e)
```

File: qemu/tests/migration.py (hunk 1 above)
Comment (dagrh) [reply qemu/tests/migration.py]: OK, see if that new version is better.

**Comment by balamuruhans** [comment]
@dagrh Thanks for the patch, let us merge it.

**Pull request merged**